libs/common-services/src/main/java/uk/gov/di/ipv/core/library/service/CriOAuthSessionService.java

```diff
@@ -17,25 +22,47 @@
 public class CriOAuthSessionService {
     private static final Logger LOGGER = LogManager.getLogger();
 
+    // Max sleeping time will be roughly WAIT_TIME * 2 ^ (MAX_ATTEMPTS - 1), plus execution time
+    // AWS say 'Consistency across all copies of data is usually reached within a second'
+    private static final int MAX_ATTEMPTS = 5;
+    private static final int WAIT_TIME_MILLIS = 50;
+
     private final DataStore<CriOAuthSessionItem> dataStore;
+    private final Sleeper sleeper;
 
-    public CriOAuthSessionService(DataStore<CriOAuthSessionItem> dataStore) {
+    public CriOAuthSessionService(DataStore<CriOAuthSessionItem> dataStore, Sleeper sleeper) {
         this.dataStore = dataStore;
+        this.sleeper = sleeper;
     }
 
     @ExcludeFromGeneratedCoverageReport
     public CriOAuthSessionService(ConfigService configService) {
         dataStore =
                 DataStore.create(
                         CRI_OAUTH_SESSIONS_TABLE_NAME, CriOAuthSessionItem.class, configService);
+        sleeper = new Sleeper();
     }
 
     public CriOAuthSessionItem getCriOauthSessionItem(String criOAuthSessionId) {
-        var criOauthSessionItem = dataStore.getItem(criOAuthSessionId);
-        if (criOauthSessionItem == null) {
-            LOGGER.warn(LogHelper.buildLogMessage("CRI OAuth session not found"));
+        try {
+            return Retry.runTaskWithBackoff(
+                    sleeper,
+                    MAX_ATTEMPTS,
+                    WAIT_TIME_MILLIS,
+                    () -> {
+                        var criOauthSessionItem = dataStore.getItem(criOAuthSessionId);
+                        if (criOauthSessionItem == null) {
+                            throw new RetryableException(new CriOAuthSessionNotFoundException());
+                        }
+                        return criOauthSessionItem;
+                    });
+        } catch (InterruptedException | NonRetryableException e) {
+            if (e instanceof InterruptedException) {
+                Thread.currentThread().interrupt();
+            }
+            LOGGER.warn(LogHelper.buildErrorMessage("Could not find CRI OAuth session", e));
+            return null;
```

Comment:
It might be preferable to just throw the exception instead - but that would require unpicking a bunch of other code :(